File: leetcode/0217-contains-duplicate/containsDuplicate.go
package leetcode

func containsDuplicate(nums []int) bool {
	var result = false
	var numMap = map[int]struct{}{}

	for _, num := range nums {
		if _, ok := numMap[num]; ok {
			result = true
			break
		}

		numMap[num] = struct{}{}
	}

	return result
}

File: leetcode/0217-contains-duplicate/containsDuplicate.ts
function containsDuplicate(nums: number[]): boolean {
  const numSet = new Set<number>();
  let result = false;

  for (const num of nums) {
    if(numSet.has(num)){
      result = true;
      break;
    }

    numSet.add(num);
  }

  return result;
}

export default containsDuplicate;
